pallets/loans/src/types/mod.rs

```diff
@@ -85,25 +85,45 @@ pub enum MutationError {
 	DiscountedCashFlowExpected,
 	/// Emits when a modification expect the loan to have an iternal pricing.
 	InternalPricingExpected,
+	/// Maturity extensions exceed max extension allowed.
+	MaturityExtendedTooMuch,
 }
 
 /// Specify the expected repayments date
 #[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
 pub enum Maturity {
 	/// Fixed point in time, in secs
-	Fixed(Moment),
+	Fixed {
+		/// Secs when maturity ends
+		date: Moment,
+		/// Extension in secs, without special permissions
+		extension: Moment,
+	},
 }
```

Comment:
Here is the migration. But it needs migration "theoretically". Right now we can purge all `pallet-loans` storage. It only requires a migration if it is used before we can deploy these changes. The same applies to all pending PRs

Reply:
Ah ok, I read you wrong 😆, I pointed to the code that causes the migration. No, there is no migration done.